frontend/ui/widget/dictquicklookup.lua

```diff
@@ -57,6 +57,66 @@ local DictQuickLookup = InputContainer:new{
     html_dictionary_link_tapped_callback = nil,
 }
 
+-- buttons available when generating custom button table
+function DictQuickLookup:stockButton(key)
+    if key == "prev_dict" then
+        return {
```

Comment:
Dunno how this could work for you, but what if:
- you let all these original buttons where they were (just for clarity of the original layout building in this file (to not have some button definitions here and others in the flow)
- instead of passing your `custom_buttons` table with string keys and real buttons, you use `ReaderDictionary:showDict(word, results, boxes, link, tweak_button_func)` and pass a function
- DictQuickLookup calls this function (if provided) after all the "nominal" buttons are built, and give it this "nominal" button table, that your `tweak_button_func` can iterate other and remove/replace (as they all have "id=", it should be easy to find/replace)

Reply:
Done :) Now only minimal change to DictQuickLookup

Reply:
Yes, thank you, that works for me (very minimal intrustion into core) if it works for you :)

Was a bit fast as suggesting a name for the func, `tweak_buttons_func` (plural) would read better.
(or `tweak_buttons_table_func` as we pass a table of buttons, but it's not the real ButtonTable widget).

Reply:
tweak_buttons_func is better. My first time experience with Lua. Dynamically typed languages are a bit of a challenge, sometimes :)